extension/scripts/content/global/ttGlobal.entry.js

```diff
@@ -1,70 +1,234 @@
 "use strict";
 
+let featureManager, globalFeatures;
+
 (async () => {
 	await loadDatabase();
 	console.log("TT: Global Entry - Loading script. ");
 
-	storageListeners.settings.push(loadGlobalEntry);
-	storageListeners.userdata.push(loadGlobalEntry);
-
-	await loadGlobalEntry();
-
-	console.log("TT: Global Entry - Script loaded.");
-})();
+	featureManager = new FeatureManager();
+	globalFeatures = new GlobalFeatures();
 
-async function loadGlobalEntry() {
-	if (settings.pages.global.alignLeft) document.documentElement.classList.add("tt-align-left");
-	else document.documentElement.classList.remove("tt-align-left");
+	globalFeatures.init();
 
-	document.documentElement.style.setProperty("--torntools-hide-upgrade-button", settings.pages.global.hideLevelUpgrade ? "none" : "block");
+	storageListeners.settings.push(function () {
+		globalFeatures.init();
+		featureManager.display(settings.featureDisplay);
+	});
+	storageListeners.userdata.push(function () {
+		globalFeatures.init();
+	});
 
-	document.documentElement.style.setProperty("--torntools-hide-leave-button", settings.pages.global.hideQuitButtons ? "none" : "flex");
+	// await loadGlobalEntry();
 
-	for (let icon of ALL_ICONS) {
-		document.documentElement.style.setProperty(`--torntools-hide-icons-${icon}`, settings.hideIcons.includes(icon) ? "none" : "initial");
-	}
+	console.log("TT: Global Entry - Script loaded.");
 
-	for (let area of ALL_AREAS.map((area) => area.class)) {
-		document.documentElement.style.setProperty(`--torntools-hide-area-${area}`, settings.hideAreas.includes(area) ? "none" : "initial");
-	}
+	requireContent().then(() => {
+		featureManager.createPopup();
+	});
+})();
 
-	// hide chats
+class GlobalFeatures {
+	constructor() {}
 
-	// clean flight
+	init() {
+		this.alignLeft();
+		this.hideLevelUpgrade();
+		this.hideLeaveButton();
+		this.hideIcons();
+		this.hideAreas();
+		// this.hideChats(); // todo
+		// this.cleanFlight(); // todo
+		this.chatFontSize();
 
-	document.documentElement.style.setProperty("--torntools-chat-font-size", `${settings.pages.chat.fontSize || 12}px`);
+		if (hasAPIData()) this.highlightRefills();
```

Comment:
Should we do these kind of checks here or in the function itself?

Reply:
Moved them to the function. Will have to test functionality just in case